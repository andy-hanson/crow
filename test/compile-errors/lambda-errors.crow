main void()
	d = () => 1
	3 foo (x string) => ()
	3 bar x => ()

foo void(x nat, f act void(nat))
	()
bar void(x nat, f act void(nat))
	()
bar void(x nat, f act void(string))
	()

