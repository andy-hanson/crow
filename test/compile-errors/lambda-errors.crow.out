exit code: 1
stderr:
test/compile-errors/lambda-errors.crow 2:9-2:16 lambda expression needs an expected type

test/compile-errors/lambda-errors.crow 3:11-3:27 the lambda doesn't match the expected type at this location.
expected type: 'act void(nat64)'

test/compile-errors/lambda-errors.crow 4:11-4:18 multiple lambda types are possible.
expected one of these types:
	'act void(nat64)'
	'act void(string)'consider explicitly typing the lambda's parameter.
